llmfoundry/models/mpt/configuration_mpt.py

```diff
@@ -222,9 +222,9 @@ def _validate_config(self) -> None:
             raise NotImplementedError(
                 'alibi only implemented with torch and triton attention.')
         if self.attn_config['attn_uses_sequence_id'] and self.attn_config[
-                'attn_impl'] not in ['torch', 'triton']:
+                'attn_impl'] not in ['torch', 'triton'] and not (self.attn_config['attn_impl']=='flash' and is_flash_v2_installed(v2_version='v2.1.2')):
             raise NotImplementedError(
-                'attn_uses_sequence_id only implemented with torch and triton attention.'
+                'attn_uses_sequence_id only implemented with torch, triton, and flash (v2.1.2 or higher) attention.'
```

Comment:
i think seq id masking should work with fa 1 as well 🤔 
why is v2.1.2 specifically required? at the very least should it be `>=2.0.0`?

Reply:
The function we use, [bert_padding.unpad_input_for_concatenated_sequences](https://github.com/Dao-AILab/flash-attention/blob/017716451d446e464dde9aca3a3c1ed2209caaa9/flash_attn/bert_padding.py#L125) was added in this [commit](https://github.com/Dao-AILab/flash-attention/commit/8f6f48d8a8c5e6c707ed3e9a2079dc1b12611122), which was added in v2.1.2.

Reply:
Should it be `>=2.1.2` then?